### PR TITLE
Small aggregator v2 cleanup

### PR DESCRIPTION
**Pull request by vusirikala**
### Description

1. Removing a few outdated TODOs.
2. Replacing a few occurrences of "Derived" with "DerivedStringSnapshot".

### CODE REVIEW

File: aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs [hunk 1]
```diff
@@ -388,8 +388,6 @@ impl<'r> TResourceGroupView for ExecutorViewWithChangeSet<'r> {
             .and_then(|g| g.inner_ops().get(resource_tag))
         {
             randomly_check_layout_matches(maybe_layout, layout.as_deref()).map_err(|e| {
-                // TODO[agg_v2](cleanup): Push into `randomly_check_layout_matches` once Satya's
```
Comment (georgemitenkov) [inline aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs:391]: We should remove this `map_err` and instead `randomly_check_layout_matches(..)?`

**Review by georgemitenkov — APPROVED** [review]

File: aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs (hunk 1 above)
Comment (vusirikala) [reply aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs:391]: Done.

**Comment by igor-aptos** [comment]
I am not sure I agree with Derived -> DerivedStringSnapshot refactor.

we can later add integer derived function, it will probably go into same Derived instance (as there is a formula applied), but will be serialized into AggregatorSnapshot. Or depending on implementation - we might need to create a new variant. 

it seems premature to change Derived -> DerivedStringSnapshot, until we see where it evolves

**Comment by vusirikala** [comment]
> I am not sure I agree with Derived -> DerivedStringSnapshot refactor.
> 
> we can later add integer derived function, it will probably go into same Derived instance (as there is a formula applied), but will be serialized into AggregatorSnapshot. Or depending on implementation - we might need to create a new variant.
> 
> it seems premature to change Derived -> DerivedStringSnapshot, until we see where it evolves

Sounds good. We can ignore this PR then.

**Comment by georgemitenkov** [comment]
@vusirikala you can still land the random layout matching fix :)

**Pull request closed without merging**